src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1084,7 +1096,7 @@ def _update_container_app_source(self, cmd, source, build_env_vars, registry_ser
         app.run_source_to_cloud_flow(source, dockerfile, build_env_vars, can_create_acr_if_needed=False, registry_server=registry_server)
 
         # Validate an image associated with the container app exists
-        containers = safe_get(self.containerapp_def, "properties", "template", "containers", default=[])
+        containers = safe_get(self.new_containerapp, "properties", "template", "containers", default=[])
```

Comment:
This is **Validate an image associated with the container app exists**, so it get existing containers from `self.containerapp_def`, which is the existing containerapp info from `client.show()`.

If there some logic can make sure the `new.containerapp` contains all existing containers?

Reply:
@Greedygre, `self.containerapp_def` is populated from `client.show()` and `new.containerapp` is populated from `self.containerapp_def` even before it reaches this code flow [here](https://github.com/Azure/azure-cli-extensions/blob/main/src/containerapp/azext_containerapp/containerapp_decorator.py#L214). So, `new_containerapp` should have all the containers? Please let me know if this is not the case.

Reply:
So it should use `self.containerapp_def` because no one will change it, but `new_containerapp` may be changed somehow, right?
What is the benefit to use `new_containerapp`?